modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/SpringSchemaBundleResourceFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.capability.xml.schema;
+
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.property.XmlModelProperty;
+import org.mule.runtime.extension.api.resources.GeneratedResource;
+
+/**
+ * Implementation of {@link AbstractXmlResourceFactory} which generates a
+ * Spring bundle file which links the extension's namespace to its schema file
+ *
+ * @since 4.0
+ */
+public class SpringSchemaBundleResourceFactory extends AbstractXmlResourceFactory
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected GeneratedResource generateXmlResource(ExtensionModel extensionModel, XmlModelProperty xmlModelProperty)
+    {
+        StringBuilder contentBuilder = new StringBuilder();
+        contentBuilder.append(getSpringSchemaBundle(xmlModelProperty, xmlModelProperty.getSchemaVersion()));
+        contentBuilder.append(getSpringSchemaBundle(xmlModelProperty, "current"));
+
+        return new GeneratedResource("spring.schemas", contentBuilder.toString().getBytes());
+    }
+
+    private String getSpringSchemaBundle(XmlModelProperty xmlProperty, String version)
+    {
+        String filename = getXsdFileName(xmlProperty);
+        return escape(String.format("%s/%s/%s=META-INF/%s\n", xmlProperty.getSchemaLocation(), version, filename, filename));
```

Comment:
The "%s/%s/%s=META-INF/%s" pattern is used in different places. Can you create a static method to build that String?